lib/nodejs/vwf.js

```diff
@@ -139,19 +141,37 @@ function HandleFileRequest( request, response ) {
 // directory structure.
 // If all that fails, serve up a 404 response since the request was not handled.
 function Serve( request, response ) {
+    lobby.post( "/sim", function( req, res ) {
+        var simDirectory = "sim";
+        var data = req.body;
+        var filePath = simDirectory + "/" + data.filePathFromSim;
+        fs.writeFile( filePath, data.simData, function( err ) {
+            if ( err ) {
+                return console.error( err );
```

Comment:
The handler should either complete the request using something like `res.sendStatus( ... )` or pass by calling `next`. The error can be forwarded to the containing application using `next( err )`. You can forward the error and exit this function by calling `return next( err )`. You can log the error here too, but it's probably fine to report it in the request, unless it relates to an unexpected error in the handler.